app/scripts/directives/service-definition-edit.js

```diff
@@ -233,6 +267,28 @@
           return svalue;
         };
 
+        var convertEnvironmentSeqOrHashToArray = function (svalue, skey) {
+          if (skey !== 'environment') {
+            return svalue;
+          }
+
+          var valueArr = [];
+          if (Array.isArray(svalue) && angular.isObject(svalue)) {
+            angular.forEach(svalue, function(lvalue) {
+              if (angular.isObject(lvalue)) {
```

Comment:
Here, if lvalue is an array like `['ENV_KEY_1=some value']` then angular.isObject(lvalue) will return true.  That doesn't appear to be what's expected based on the comments in the code.